.github/workflows/linters.yml

```diff
@@ -4,6 +4,51 @@ on:
 permissions:
   contents: read
 jobs:
+  lint-golangci:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+          check-latest: true
+
+      - name: lint code
+        run: |
+          make GOLANGCI_LINT_FIX=false GOLANGCI_LINT_BASE_REV=HEAD~ lint-code
+
+      - name: check-is-dirty
+        run: |
+          if [[ -n $(git status --porcelain) ]]; then
+            echo "Detected uncommitted changes."
+            git status
+            git diff
+            exit 1
+          fi
+
+      - name: check-no-assertions
+        run: |
+          make check-no-assertions-present
```

Comment:
I moved this here from `linters.yml` since I can't see a reason why this would need to be separate.